src/tr1/game/objects/creatures/rat.c

```diff
@@ -208,15 +220,8 @@ static void M_ControlVole(const int16_t item_num)
 
         Item_Animate(item);
 
-        int32_t wh = Room_GetWaterHeight(
-            item->pos.x, item->pos.y, item->pos.z, item->room_num);
-        if (wh == NO_HEIGHT) {
-            item->object_id = O_RAT;
-            item->current_anim_state = RAT_STATE_DEATH;
-            item->goal_anim_state = RAT_STATE_DEATH;
-            Item_SwitchToAnim(item, RAT_DIE_ANIM, -1);
-            item->pos.y = item->floor;
-        }
+        int32_t water_height;
+        Creature_EnsureHabitat(item_num, &water_height, &m_RatInfo);
```

Comment:
Wdyt about making water height a nullptr for those cases?

Reply:
Thanks, done.